Change tooltip position when dropdown is opened

IshavSohal opened the pull request:
### Related Item(s)
#2337

### Changes
- Made the dropdown trigger's tooltip trigger manually, so that its position can be adjusted before it's opened
  - If the dropdown is open, the tooltip will use the `tooltipPositionAlt` prop as its position
  - If the dropdown isn't open, the tooltip will use the `tooltipPosition` prop as its position
- When a dropdown is opened, the dropdown trigger's tooltip will disappear, so that it doesn't block the dropdown (especially useful when in keyboard focus)

### Note
- Below I have listed the testing steps for a few dropdown menus, but you can perform similar steps on the other dropdown menus:
  - Notifications dropdown
  - About RAMP dropdown
  - Datagrid options dropdown
  - Datagrid hide columns dropdown
  - Export settings button
  - Legend layer options button dropdown(s)

### QA Testing
Please test against main branch https://ramp4-pcar4.github.io/ramp4-pcar4/main/demos/enhanced-samples.html

### Testing
Steps:
1. Open the classic main sample
##### Language dropdown
2. Press Tab until you have reached the 'Change Language' button
3. Click Enter, and observe that the tooltip has disappeared. 
4. Hover over the  'Change Language' button with your mouse, and observe that the tooltip appears to the left
5. Perform these testing steps in `main` and observe that the tooltip covers the dropdown when it's open. 
##### Toggle Groups dropdown
6. Press Shift+Tab until you reach the legend panel
7. Press Space/Enter to enter it, and press Tab until you reach the 'Toggle Groups' button
8. Click Enter, and observe that the tooltip has disappeared. 
9.  Hover over the  'Toggle Groups' button with your mouse, and observe that the tooltip appears on the bottom
10. Perform these testing steps in `main` and observe that the tooltip covers the dropdown when it's open. 
##### Toggle Visibility dropdown
11. Press Tab until you reach the 'Toggle Visibility' button
12. Click Enter, and observe that the tooltip has disappeared. 
13.  Hover over the  'Toggle Visibility' button with your mouse, and observe that the tooltip appears on the bottom
14. Perform these testing steps in `main` and observe that the tooltip covers the dropdown when it's open. 

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/ramp4-pcar4/ramp4-pcar4/2357)
<!-- Reviewable:end -->

mohsin-r reviewed (approved):
> Reviewed 7 of 7 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ramp4-pcar4/ramp4-pcar4/2357)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @IshavSohal)



<!-- Sent from Reviewable.io -->

szczz reviewed (approved):
Ultra-minor grouse, but figured I'd bring it to your attention. It seems like the tooltip will jump around on language selection based on mouse placement. 

![changin.gif](https://files.reviewableusercontent.io/images/reviewable/28012494/xUs0KFi2hxlVCRwhIny8g9zZ/changin.gif)

> Reviewed 7 of 7 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ramp4-pcar4/ramp4-pcar4/2357#-:-O66bqkH07el--DMeBOa:b-ccb6pe)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @IshavSohal)



<!-- Sent from Reviewable.io -->

milespetrov reviewed (changes requested):
> Reviewed 7 of 7 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ramp4-pcar4/ramp4-pcar4/2357)* status: all files reviewed, 1 unresolved discussion (waiting on @IshavSohal)

___
*[`src/components/controls/dropdown-menu.vue` line 80 at r1](https://reviewable.io/reviews/ramp4-pcar4/ramp4-pcar4/2357#-O676d264nU6cIo6g-JH:-O676d264nU6cIo6g-JI:b-qnp7k0) ([raw file](https://github.com/ramp4-pcar4/ramp4-pcar4/blob/c068a8cbcebd6b25ec0cb5f1023a6033ea2d43ee/src/components/controls/dropdown-menu.vue#L80)):*
> ```Vue
> const toggleDropdown = () => {
>     open.value = !open.value;
>     (dropdownTrigger.value as any)._tippy.hide();
> ```

It would be nice to have proper types if possible. Is `._tippy.hide()` the official way to hide a tooltip or some workaround we need to use. I can see there being no types for private members.


<!-- Sent from Reviewable.io -->

IshavSohal commented:
Updated Enhanced Testing:
Samples: https://ishavsohal.github.io/ramp4-pcar4/issue-2337-fix/demos/enhanced-samples.html
Catalogue: https://ishavsohal.github.io/ramp4-pcar4/issue-2337-fix/demos/enhanced-all.html

Updated Legacy Testing:
Main: https://ishavsohal.github.io/ramp4-pcar4/issue-2337-fix/
Catalogue: https://ishavsohal.github.io/ramp4-pcar4/issue-2337-fix/demos/index-all.html
Samples: https://ishavsohal.github.io/ramp4-pcar4/issue-2337-fix/demos/index-samples.html

james-rae reviewed (approved):
> Reviewed 7 of 7 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ramp4-pcar4/ramp4-pcar4/2357)* status: all files reviewed, 1 unresolved discussion (waiting on @milespetrov)

___
*[`src/components/controls/dropdown-menu.vue` line 80 at r1](https://reviewable.io/reviews/ramp4-pcar4/ramp4-pcar4/2357#-O676d264nU6cIo6g-JH:-O6Myh0M00qCmBLWVzQi:b-gy7411) ([raw file](https://github.com/ramp4-pcar4/ramp4-pcar4/blob/085decab27e8ee39a808b8373ac8ca041e4e9554/src/components/controls/dropdown-menu.vue#L80)):*
<details><summary><i>Previously, milespetrov (Miles Petrov) wrote…</i></summary><blockquote>

It would be nice to have proper types if possible. Is `._tippy.hide()` the official way to hide a tooltip or some workaround we need to use. I can see there being no types for private members.
</blockquote></details>

I'm no expert but can see that type of `(any)._tippy.foo()` notation peppered all over the codebase. Could be that tippy operates by putting that `_tippy` on DOM objects but i'm just handwaving.


<!-- Sent from Reviewable.io -->

james-rae reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/ramp4-pcar4/ramp4-pcar4/2357)* status: all files reviewed, 1 unresolved discussion (waiting on @milespetrov)



<!-- Sent from Reviewable.io -->
